packages/shared/src/features/annotation/types.ts

```diff
@@ -13,3 +13,92 @@ export type ConfigCategory = z.infer<typeof configCategory>;
 export type CastedConfig = Omit<ScoreConfig, "categories"> & {
   categories: ConfigCategory[] | null;
 };
+
+const ScoreSource = z.enum(["ANNOTATION", "API", "EVAL"]);
+
+const ScoreBase = z.object({
+  id: z.string(),
+  timestamp: z.date(),
+  projectId: z.string(),
+  name: z.string(),
+  source: ScoreSource,
+  authorUserId: z.string().nullish(),
+  comment: z.string().nullish(),
+  traceId: z.string(),
+  observationId: z.string().nullish(),
+  configId: z.string().nullish(),
+  createdAt: z.date(),
+  updatedAt: z.date(),
+});
+
+const NumericData = z.object({
+  value: z.number(),
+  stringValue: z.undefined().nullish(),
+  dataType: z.literal("NUMERIC"),
+});
+
+const CategoricalData = z.object({
+  value: z.number().optional().nullish(),
+  stringValue: z.string(),
+  dataType: z.literal("CATEGORICAL"),
+});
+
+const BooleanData = z.object({
+  value: z.number(),
+  stringValue: z.string(),
+  dataType: z.literal("BOOLEAN"),
+});
+
+export const ScoreUnion = z.discriminatedUnion("dataType", [
+  ScoreBase.merge(NumericData),
+  ScoreBase.merge(CategoricalData),
+  ScoreBase.merge(BooleanData),
+]);
+
+const GetAllScoresBase = z.object({
+  id: z.string(),
+  timestamp: z.date(),
+  name: z.string(),
+  source: ScoreSource,
+  comment: z.string().nullish(),
+  traceId: z.string(),
+  observationId: z.string().nullish(),
+  trace: z.object({
+    userId: z.string(),
+  }),
+});
+
+export const GetAllScores = z.discriminatedUnion("dataType", [
+  GetAllScoresBase.merge(NumericData),
```

Comment:
how dies getall work with value but without an operator?

Reply:
I don't understand your question, could you rephrase please?